src/core/Track.cpp

```diff
@@ -184,32 +184,24 @@ bool TrackContentObject::comparePosition(const TrackContentObject *a, const Trac
 
 
 
-/*! \brief Copy this TrackContentObject to the clipboard.
+/*! \brief Copies the state of a TrackContentObject to another TrackContentObject
  *
- *  Copies this track content object to the clipboard.
+ *  This method copies the state of a TCO to another TCO
  */
-void TrackContentObject::copy()
+void TrackContentObject::copyStateTo( TrackContentObject *src, TrackContentObject *dst )
 {
-	Clipboard::copy( this );
-}
-
-
-
+	QDomDocument doc;
+	QDomElement parent = doc.createElement( "StateCopy" );
+	src->saveState( doc, parent );
 
-/*! \brief Pastes this TrackContentObject into a track.
- *
- *  Pastes this track content object into a track.
- *
- * \param _je The journal entry to undo
- */
-void TrackContentObject::paste()
-{
-	if( Clipboard::getContent( nodeName() ) != NULL )
+	// If the node names match we copy the state
+	if( src->nodeName() == dst->nodeName() )
```

Comment:
I think this if statement can surround the whole of this method. There's no need to serialise the state if it's not going to be deserialised, and there's no need to resolve and update things if nothing has changed.